src/ToolBox/superpmi/superpmi/superpmi.cpp

```diff
@@ -45,9 +45,20 @@ void SetSuperPmiTargetArchitecture(const char* targetArchitecture)
         SpmiTargetArchitecture = SPMI_TARGET_ARCHITECTURE_AMD64;
     }
 #elif defined(_TARGET_X86_)
-    SpmiTargetArchitecture = SPMI_TARGET_ARCHITECTURE_X86;
+    if ((targetArchitecture != nullptr) && (0 == _stricmp(targetArchitecture, "arm")))
+    {
+        SpmiTargetArchitecture = SPMI_TARGET_ARCHITECTURE_ARM;
+    }
+    else
+    {
+        SpmiTargetArchitecture = SPMI_TARGET_ARCHITECTURE_X86;
+    }
 #elif defined(_TARGET_ARM_)
     SpmiTargetArchitecture = SPMI_TARGET_ARCHITECTURE_ARM;
+#elif defined(_TARGET_ARM64)
+    SpmiTargetArchitecture = SPMI_TARGET_ARCHITECTURE_ARM64
+#else
+#error Unsupported target architecture
```

Comment:
Is it unsupported target or host architecture?

Reply:
In this particular case it's both 🤣 I will change the error to be just "Unsupported architecture"